src/rtcp/packet.ts

```diff
@@ -0,0 +1,204 @@
+/*
+        0                   1                   2                   3
+        0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+header |V=2|P|    SC   |      PT       |             length            |
+       +=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+
+ */
+
+/** @ignore */
+const RTCP_VERSION = 2;
+/** @ignore */
+const COMMON_HEADER_LENGTH = 4;
+
+export enum PacketType
+{
+	SR = 200,
+	RR = 201,
+	SDES = 202,
+	BYE = 203,
+	APP = 204,
+	RTPFB = 205,
+	PSFB = 206,
+	XR = 207
+}
+
+/**
+ * Inspect the given buffer and return a boolean indicating whether it could be
+ * a valid RTCP packet or not.
+ *
+ * ```ts
+ * if (isRtcp(buffer)) {
+ *   console.log('it seems a valid RTCP packet');
+ * }
+ * ```
+ */
+export function isRtcp(buffer: Buffer): boolean
+{
+	const firstByte = buffer.readUInt8(0);
+
+	return (
+		Buffer.isBuffer(buffer) &&
+		buffer.length >= COMMON_HEADER_LENGTH &&
+		// DOC: https://tools.ietf.org/html/draft-ietf-avtcore-rfc5764-mux-fixes
+		(firstByte > 127 && firstByte < 192) &&
+		// RTCP Version must be 2.
+		(firstByte >> 6) === RTCP_VERSION &&
+		// RTCP packet types defined by IANA:
+		// http://www.iana.org/assignments/rtp-parameters/rtp-parameters.xhtml#rtp-parameters-4
+		// RFC 5761 (RTCP-mux) states this range for secure RTCP/RTP detection.
+		(buffer.readUInt8(1) >= 192 && buffer.readUInt8(1) <= 223)
+	);
+}
+
+export abstract class RtcpPacket
+{
+	// Buffer.
+	// @ts-ignore. 'buffer' has not initializer and is not assigned in constructor.
+	protected buffer: Buffer;
+	// RTCP Packet type.
+	private packetType: PacketType;
+
+	/**
+	 * @ignore
+	 *
+	 * @param Buffer.
+	 */
+	static getCount(buffer: Buffer): number
+	{
+		return buffer.readUInt8(0) & 0x1F;
+	}
+
+	/**
+	 * @ignore
+	 *
+	 * @param PacketType.
+	 */
+	protected constructor(packetType: PacketType)
+	{
+		this.packetType = packetType;
+	}
+
+	/**
+	 * @ignore
+	 */
+	dump(): any
+	{
+		return {
+			version : this.getVersion(),
+			count   : this.getCount(),
+			length  : this.getLength(),
+			padding : this.getPadding()
+		};
+	}
+
+	/**
+	 * Get the internal buffer containing the serialized RTCP binary packet.
+	 */
+	abstract getBuffer(): Buffer;
+
+	/**
+	 * Get the RTCP version of the packet (always 2).
+	 */
+	getVersion(): number
+	{
+		return RTCP_VERSION;
+	}
+
+	/**
+	 * Get the padding flag.
+	 */
+	getPadding(): boolean
```

Comment:
I don't like this. This is too RFC based. In `RtpPacket`, the `getPadding()` method returns the size (in bytes) of the padding. Same for `setPadding()`. My initial idea was to make this a friendly API rather than a strict RFC based API. For example: I don't like the idea of setting the `padding` bit to 1 and then do not add real padding.